src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -1321,6 +1322,86 @@ internal static object ContainsOperator(ExecutionContext context, IScriptExtent
             return BoolToObject(!contains);
         }
 
+        internal static bool AnyOperator(
+            ExecutionContext context,
+            CallSite<Func<CallSite, object, IEnumerator>> getEnumeratorSite,
+            CallSite<Func<CallSite, object, object, object>> comparerSite,
+            object left,
+            object right)
+        {
+            ScriptBlock predicate = right as ScriptBlock;
+            if (predicate == null)
+            {
+                return ContainsOperatorCompiled(context, getEnumeratorSite, comparerSite, left, right);
+            }
```

Comment:
This feels like it'd be better as pattern matching, but in truth it looks weird either way so 🤷‍♂

Reply:
How about:

```csharp
if (!(right is ScriptBlock predicate))
    return ContainsOperatorCompiled(...)
```